src/serac/physics/materials/thermal_functional_material.hpp

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file thermal_functional_material.hpp
+ *
+ * @brief The material and load types for the thermal functional physics module
+ */
+
+#pragma once
+
+#include "serac/numerics/functional/functional.hpp"
+
+/// ThermalConductionFunctional helper structs
+namespace serac::Thermal {
+
+/// Linear isotropic thermal conduction material model
+struct LinearIsotropicConductor {
+  /// Density
+  double density_;
+
+  /// Specific heat capacity
+  double specific_heat_capacity_;
+
+  /// Constant isotropic thermal conductivity
+  double conductivity_;
```

Comment:
Possibly naive question, is it worth doing bad data checks to ensure that the material properties are non-negative to ensure we obey conversation laws?